### PR TITLE
small fix in HAL builder caused by TypeName

### PR DESCRIPTION
**Pull request by marco-scoppetta**
- and fixed test to avoid regression

### CODE REVIEW

File: grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java [hunk 1]
```diff
@@ -221,4 +228,22 @@ private static void attachSingleGeneratedRelation(Representation currentHal, Con
             throw new RuntimeException("Unrecognized base type of " + type);
         }
     }
+
+    static void generateConceptState(Representation resource, Concept concept){
+
+        if (concept.isInstance()) {
+            Instance instance = concept.asInstance();
+            resource.withProperty(ID_PROPERTY, instance.getId().getValue())
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java]: I think it's confusing that in the case of an instance the ID property is filled with a concept's ID

File: grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java [hunk 2]
```diff
@@ -221,4 +228,22 @@ private static void attachSingleGeneratedRelation(Representation currentHal, Con
...
+                    .withProperty(TYPE_PROPERTY, instance.type().getName().getValue())
+                    .withProperty(BASETYPE_PROPERTY, getBaseType(instance).name());
+        } else {
+            Type type = concept.asType();
+            resource.withProperty(ID_PROPERTY, type.getName().getValue())
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java]: But in the case of a Type it's filled with a Type's Name. To limit confusion I propose that the ID should be the concept's ID regardless of it being a type or instance and a new property Name should be included in the resource for Types.

**Comment by fppt** [comment]
Changing the above will probably require some changes in the dashboard. So do let me know how feasible these changes are.

**Comment by marco-scoppetta** [comment]
I completely agree with this, I also wanted to propose this change but never did eventually.
This will require a bit of changes in Engine and Dashboard, I guess I can start doing the changes tomorrow.

**Comment by fppt** [comment]
My comments will be addressed in a future PR. Work detailed here: https://work.grakn.ai/entity/12507

**Pull request merged**